lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -1133,6 +1269,50 @@ push @tests, __PACKAGE__->new(
         table => 'ap',
       appname => 'sql-ledger',
   min_version => '2.7',
+  max_version => '2.8'
+);
+
+    push @tests, __PACKAGE__->new(
+        # Add a unique key to allow editing
+        # Make the test serially reusable and protect against triggers
+        test_query => "SELECT ac.i_key, ac.trans_id, ac.id, ac.memo, ac.amount, xx.description, 
+                              ch.accno, ch.link, ch.charttype, ch.category, ac.cleared, approved
+                         FROM acc_trans ac 
+                         JOIN (
+                                   SELECT g.id, g.description FROM gl g
+                             UNION SELECT a.id, n.name        FROM ar a JOIN customer n ON n.id = a.customer_id
+                             UNION SELECT a.id, n.name        FROM ap a JOIN vendor n   ON n.id = a.vendor_id
+                         ) xx ON xx.id = ac.trans_id
+                         JOIN chart ch ON (ac.chart_id = ch.id)
+                        WHERE ( ch.category NOT IN ( 'A', 'L', 'Q' ) 
+                             OR ch.link NOT LIKE '%paid' ) 
+                          AND ac.cleared IS NOT NULL 
+                          AND ac.approved
+                     ORDER BY accno, transdate, ac.id",
+      display_name => marktext('Reconciliations on non-bank accounts'),
```

Comment:
While I agree that this could be a sensible data integrity check, I'm wondering: does this situation prevent running an acceptable migration?

Reply:
Nope, the user can force the migration even with those still present.

Reply:
Or skip this test completely